homeassistant/components/sensor/ecoal_boiler.py

```diff
@@ -0,0 +1,83 @@
+"""
+Allows reading temperatures from ecoal/esterownik.pl controller.
+
+For more details about this platform, please refer to the documentation at
+https://home-assistant.io/components/sensor.ecoal_boiler/
+"""
+
+import voluptuous as vol
+
+from homeassistant.components.switch import PLATFORM_SCHEMA
+from homeassistant.components.ecoal_boiler import DATA_ECOAL_BOILER
+from homeassistant.const import TEMP_CELSIUS
+from homeassistant.helpers.entity import Entity
+import homeassistant.helpers.config_validation as cv
+
+# Available temp sensor ids
+SENSOR_IDS = (
+    "outdoor_temp",
+    "indoor_temp",
+    "indoor2_temp",
+    "domestic_hot_water_temp",
+    "target_domestic_hot_water_temp",
+    "feedwater_in_temp",
+    "feedwater_out_temp",
+    "target_feedwater_temp",
+    "coal_feeder_temp",
+    "exhaust_temp",
+)
+
+ENABLED_SCHEMA = {}
+for tempsensor_id in SENSOR_IDS:
+    ENABLED_SCHEMA[vol.Optional(tempsensor_id)] = cv.string
+CONF_ENABLE = "enable"
+PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend(
```

Comment:
Instead of specifying a config section for each platform, keep the all the config in the component section, and load the platforms via discovery from the component.

Look at the octoprint component for the correct schema structure.

The platform schema here can be removed.

Reply:
You want me to move to parse pump/sensor configuration to component config and parse it in setup(hass, config) and call load_platform() for parsed pumps and sensors configs?
It is just moving configuration to different part of yaml, or I am missing other points here?

Reply:
Yes, basically.

Having all of the config in the component section is the best option for components with platforms.

Reply:
Done.